e2e/config/config.go

```diff
@@ -386,18 +396,29 @@ func validateDuplicateDeployerNames(deployers []Deployer) error {
 
 // validateDuplicateDeployerContent ensures no two deployers have the same type and content
 func validateDuplicateDeployerContent(deployers []Deployer) error {
-	seen := make(map[Deployer]Deployer)
+	type LocalDeployer struct {
+		Type   string  `json:"type"`
+		Recipe *Recipe `json:"recipe,omitempty"`
+	}
+
+	seen := make(map[string]LocalDeployer)
 
 	for _, deployer := range deployers {
-		key := Deployer{
-			Type: deployer.Type,
+		value := LocalDeployer{
+			Type:   deployer.Type,
+			Recipe: deployer.Recipe,
 		}
 
-		if duplicate, exists := seen[key]; exists {
-			return fmt.Errorf("duplicate deployer content found:\n	%+v\n	%+v", duplicate, deployer)
+		for existingDeployerName, existingDeployer := range seen {
+			if reflect.DeepEqual(value, existingDeployer) {
+				return fmt.Errorf("duplicate deployer content found: "+
+					"%v is same as %v\n	"+
+					"%+v\n	%+v",
+					existingDeployerName, deployer.Name, existingDeployer, value)
+			}
 		}
 
-		seen[key] = deployer
+		seen[deployer.Name] = value
```

Comment:
We can simplify this check to:

```go
We can simplify like this:

```go
    type content struct {
		type   string
		recipe Recipe
    }
   
    seen := make(map[content]Deployer])

    for _, deployer := range deployers {
        key := content{type: deployer.Type}
        if deployer.Recipe != nil {
            key.recipe = *deployer.Recipe
        }

        if duplicate, ok := seen[key]; ok {
            fail...
        }
        
        seen[key] = deployer
    }
```

Reply:
Done